host/tests/gatt_derive.rs

```diff
@@ -137,11 +108,34 @@ async fn gatt_client_server() {
                         adv_data: &adv_data[..],
                         scan_data: &scan_data[..],
                     }).await?;
-                    let _conn = acceptor.accept().await?;
+                    let conn = acceptor.accept().await?;
                     println!("[peripheral] connected");
-                    // Keep it alive
+                    let mut writes = 0;
                     loop {
-                        tokio::time::sleep(Duration::from_secs(10)).await;
+                        match conn.next().await {
+                            ConnectionEvent::Disconnected { reason } => {
+                                println!("Disconnected: {:?}", reason);
+                                break;
+                            }
+                            ConnectionEvent::Gatt { connection: _, event } => match event {
+                                GattEvent::Write {
+                                    value_handle: handle
+                                } => {
+                                    let characteristic = server.server().table().find_characteristic_by_value_handle(handle).unwrap();
+                                    let value: u8 = server.server().table().get(&characteristic).unwrap();
+                                    assert_eq!(expected, value);
+                                        expected = expected.wrapping_add(1);
```

Comment:
I think you'll need to add 2 here - in gatt_derive the first write is rejected by the on_write callback of the characteristic (to check the mechanism is working)

Reply:
Ah, thanks, I blindly copied this from the other test.

Reply:
Hmm, I'm not sure if it's correct to change it, the old test was already doing a +=1

Reply:
Apologies, you are correct, I was looking at the writes, not expected :man_facepalming: